File: cpp/symmetric_tree.cpp
#include <stdio.h>
#include <vector>
#include <queue>

using namespace std;

/**
 * Definition for binary tree
 */
struct TreeNode {
     int val;
     TreeNode *left;
     TreeNode *right;
     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 };

class Solution {
public:
	bool isTwoSymmetric(TreeNode *l, TreeNode *r)
	{
		if (l == NULL && r == NULL)
		{
			return true;
		}
		else if (l == NULL || r == NULL)
		{
			return false;
		}
		else
		{
			return ((l->val == r->val) && isTwoSymmetric(l->left, r->right) && isTwoSymmetric(l->right, r->left));
		}
	}

    bool isSymmetric(TreeNode *root) {
    	if (root == NULL) return true;

   		return isTwoSymmetric(root->left, root->right);
    }
};

int main()
{
	return 0;
}

File: python/symmetric-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSymmetric(self, root: TreeNode) -> bool:
        def isSymmetric(l, r):
            if not l and not r: return True
            elif (l and not r) or (not l and r): return False
            else:
                return l.val==r.val and isSymmetric(l.left, r.right) and isSymmetric(l.right, r.left)
            
        if not root: return True
        return isSymmetric(root.left, root.right)

